src/Components/BehaviorsModal/Internal/Widgets/DataHistoryWidget/Internal/DataHistoryWidgetErrorHandling.tsx

```diff
@@ -0,0 +1,188 @@
+        let details = '';
+
+        if (requestStatus === 0) {
+            // e.g. with network error of ERR_NAME_NOT_RESOLVED due to invalid request url (including cluster url error) or connection error
+            description = t('widgets.dataHistory.errors.genericNetwork');
+            imgSrc = ConnectionErrorImg;
+        } else if (requestStatus === 400) {
+            switch (response.data?.error.code) {
+                case DataHistoryErrors.General_BadRequest: // query error (including table name since it is part of query in the request payload)
+                    description = t(
+                        'widgets.dataHistory.errors.generalBadRequestMessage'
+                    );
+                    imgSrc = GenericErrorImg;
+                    break;
+                case DataHistoryErrors.BadRequest_EntityNotFound: // database error
+                    description = t(
+                        'widgets.dataHistory.errors.entityNotFound'
+                    );
+                    imgSrc = ConnectionErrorImg;
+                    break;
+                default:
+                    description = t(
+                        'widgets.dataHistory.errors.genericBadRequest'
+                    );
+                    imgSrc = GenericErrorImg;
+                    break;
+            }
+            details = response.data?.error.innererror?.message;
+        } else if (requestStatus === 403) {
+            // permission error
+            description = t('widgets.dataHistory.errors.forbidden');
+            imgSrc = PermissionErrorImg;
+            details = response.data?.error.innererror?.message;
+        } else {
+            imgSrc = GenericErrorImg;
+            description = t('widgets.dataHistory.errors.genericBadRequest');
+        }
+        return { description, imgSrc, details };
+    }, [requestStatus, response]);
+
+    const theme = useTheme();
+    const classNames = getDataHistoryWidgetClassNames(styles, { theme });
+    const sharedCalloutStyles = getCardboardListCalloutStyles(theme);
+
+    const errorDetailsCalloutId = useId('show-error-details-button');
+    const errorDetailsCalloutDescriptionId = useId(
+        'error-details-callout-description'
+    );
+    const errorDetailsCallout = useMemo(() => {
+        if (response?.data?.error.innererror?.message) {
+            return (
+                isDetailsCalloutVisible && (
+                    <Callout
+                        ariaDescribedBy={errorDetailsCalloutDescriptionId}
+                        styles={{
+                            ...sharedCalloutStyles,
+                            root: {
+                                ...(sharedCalloutStyles.root as any),
+                                fontSize: 12
+                            }
+                        }}
+                        target={`#${errorDetailsCalloutId}`}
+                        onDismiss={toggleIsDetailsCalloutVisible}
+                        role="alert"
```

Comment:
what does setting this role do? I haven't seen us doing this much in our repo, what made you decide to do it here?

Reply:
it was in fluent example for the Callout, it sends the alert to the screen reader and the text content is being read automatically when the callout is opened.

Reply:
very cool. Great find.